src/NexusMods.Paths/AbsolutePath.cs

```diff
@@ -148,17 +234,17 @@ public void GetFullPath(Span<char> buffer)
 
         Directory.CopyTo(buffer);
 
-        if (string.IsNullOrEmpty(FileName)) return;
+        // directory without file name
+        if (FileName.Length == 0) return;
 
-        // on Linux: Directory="/", FileName="foo" should return "/foo" and not "//foo"
-        if (OperatingSystem.IsWindows() || Directory != DirectorySeparatorCharStr)
+        if (IsRootDirectory(Directory))
```

Comment:
I chose to use `IsRootDirectory` instead of `Directory.EndsWith`, as it better conveys, what is being checked. We handle root directories in a special way, since they end with a directory separator character. However, if we change the code to just check for that separator character, it's not really clear that this very specific condition, is for root directories.

You can argue that this would be solved by adding a comment, but then you have two sources of truth, and we might end up in a similar situation as before, where the handling of root directories is kinda broken.

Reply:
I was just curious from a performance standpoint really. Ideally the tests should be comprehensive enough to catch that kind of error should it ever occur.

Reply:
Thinking about it, `GetFullPathWithSeparator()` should probably be deprecated; as `GetFullPath()` was reworked to guarantee a separator for root paths [not having it was originally by design, as perf. measure].

Reply:
> https://github.com/Nexus-Mods/NexusMods.App/pull/187#discussion_r1135805350

I'd prefer standardising to the approach that is faster at runtime since `GetFullPath()` is a high activity API.

And yeah, `GetFullPathWithSeparator` can now just be replaced with `GetFullPath` since we now guarantee a separator at the end of root path.

Reply:
Moved to issue https://github.com/Nexus-Mods/NexusMods.App/issues/189